src/pymor/models/iosys.py

```diff
@@ -789,15 +791,21 @@ def _gramian(self, typ, mu=None):
         elif typ == 'lqg_o_lrcf':
             return solve_ricc_lrcf(A, E, B.as_range_array(mu=mu), C.as_source_array(mu=mu),
                                    trans=True, options=options_ricc_lrcf)
+        elif typ == 'pr_c_lrcf':
```

Comment:
I think we should mention in the docstring for the `pr_c_lrcf` and `pr_o_lrcf` Gramians that $D+D^T$ must be invertible, and maybe add an assertion like
```python
assert not isinstance(D, ZeroOperator), "D+D^T must be invertible."
```
Alternatively, we might opt to automatically add a perturbation similar to https://github.com/pymor/pymor/blob/dfe7c32dbb4b92b8128c74837cb65a13e248d970/src/pymordemos/phlti.py#L131, and maybe log a warning.

Reply:
For now, I would just add a note in the documentation.